### PR TITLE
We are missing 4 bytes in the LZ4 compression buffer.

### PR DESCRIPTION
**Pull request by fulmicoton**
Closes #831

### CODE REVIEW

**Review by PSeitz — APPROVED** [review]
The 4 bytes are already included in get_maximum_output_size, but let's not rely on that

File: src/store/compression_lz4_block.rs [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
 
 use core::convert::TryInto;
+use std::mem;
 use lz4_flex::{compress_into, decompress_into};
 
 #[inline]
 pub fn compress(uncompressed: &[u8], compressed: &mut Vec<u8>) -> io::Result<()> {
     compressed.clear();
-    let maximum_ouput_size = lz4_flex::block::get_maximum_output_size(uncompressed.len());
+    let maximum_ouput_size = mem::size_of::<u32>() + lz4_flex::block::get_maximum_output_size(uncompressed.len());
```
Comment (PSeitz) [inline src/store/compression_lz4_block.rs:11]: reserve 4 bytes more

Comment (fulmicoton) [reply src/store/compression_lz4_block.rs:11]: I don't understand this comment?

**Comment by fulmicoton** [comment]
> The 4 bytes are already included in get_maximum_output_size, but let's not rely on that

The previous code was creating a slice with unallocated memory. (in addition to being uninitialized)

**Comment by PSeitz** [comment]
> 
> The previous code was creating a slice with unallocated memory. (in addition to being uninitialized)

With the `reserve` call, it should be allocated?
Uninitialized data is being handled in lz4_flex via `MaybeUninit`.

**Comment by fulmicoton** [comment]
@PSeitz  Now it is allocated. Before this PR, the last 4 bytes were not necessarily allocated.
If you know how to deal with the MaybeUninitialized thing that would be awesome to remove the allow clippy thing.

**Pull request merged**